Modernize and enhance 24-hour clock support

penrods opened the pull request:
Several changes to modernize this skill:
* Display of time is now controlled by a Skill Setting
* Add nice_time(), which formats 24-hour time better (e.g. "13 hundred")
* Add query for date or day of week
* Improve pronunciation by changing "timezone" to "time zone"
* Switched from Timer to using a repeating event
* Removed leading '0' from time display for 12-hour clock
* Update license to Apache 2.0, add .gitignore
* Removed unused libraries
* Update README
* Use new naming standards for .voc files

Commit:
Modernize and enhance 24-hour clock support

Several changes to modernize this skill:
* Display of time is now controlled by a Skill Setting
* Add nice_time(), which formats 24-hour time better (e.g. "13 hundred")
* Add query for date or day of week
* Improve pronunciation by changing "timezone" to "time zone"
* Switched from Timer to using a repeating event
* Removed leading '0' from time display for 12-hour clock
* Update license to Apache 2.0, add .gitignore
* Removed unused libraries
* Update README
* Use new naming standards for .voc files

penrods commented:
This should also resolve #1 and #2

forslund commented:
Just started the review something isn't working quite right. When asked about the time, Mycroft answers "It's one thousand one hundred twenty five". (24 hour clock)

Commit:
Fix pronunciation of time "10:24"

Times such as "10:24" were being pronounced as "one thousand and 24".
Added the colon to allow Mimic to correct this.

forslund commented on __init__.py:
```diff
@@ -1,85 +1,128 @@
-LOGGER = getLogger(__name__)
+    if use_24hour:
+        # e.g. "03:01" or "14:22"
+        str = dt.strftime("%H:%M")
```
Not really an issue but consider using `string` instead if `str` since str is the builtin string type.

forslund commented on __init__.py:
```diff
@@ -1,85 +1,128 @@
+                # TODO: convert "23" to "twenty three" in helper method
+
+                # Mimic is speaking "23 34" as "two three 43" :(
+                # but it does say "2343" correctly.  Not ideal for general
```
This is slightly incorrect

forslund reviewed (commented):
Minor nit-picks

Commit:
Fix screen clearing

The screen-clearing was leaving some pixels behind.  Expanded the
blank areas to fix.

Commit:
Fix review items

Minor fixes:
* Change variable name self.active to be more clear
* Added some comments to explain behavior
* Improved formatting in nice_time() -- temporary until final
  version becomes part of mycroft-core
